orttraining/orttraining/training_api/include/checkpoint_property.h

```diff
@@ -4,63 +4,40 @@
 #pragma once
 
 #include <type_traits>
+#include <variant>
+
 #include "onnx/defs/tensor_proto_util.h"
 
 namespace onnxruntime {
 namespace training {
 namespace api {
 
-template <typename T>
-struct TypedCheckpointProperty;
+typedef std::variant<int64_t, float, std::string> CheckPointPropertyDataType;
 
 /**
- * @brief Base class for user defined checkpoint property.
+ * @brief Class for user defined checkpoint property.
  */
 struct CheckpointProperty {
```

Comment:
do we need a separate CheckpointProperty struct? could PropertyBag just contain `unordered_map<string, CheckPointPropertyDataType>`?